src/coreclr/jit/codegenarm64.cpp

```diff
@@ -4725,7 +4725,6 @@ void CodeGen::genCodeForCinc(GenTreeOp* cinc)
     regNumber               targetReg = cinc->GetRegNum();
     regNumber               srcReg    = op->IsIntegralConst(0) ? REG_ZR : genConsumeReg(op);
 
-    assert(op->IsCnsIntOrI());
```

Comment:
This is the assert getting triggered for the failing spmi replays.

We can remove this assert as we have an explicit guard to check for op being constant before generating a CINC/CINCC node.
The stress tests create a scenario where the constant int in the op is spilled to the stack. The RELOAD node doesn't convey the constantness info, causing the assert to trigger.

The `spmi/repro.libraries_tests.pmi.linux.arm64.checked/repro-269819.mc` generated the following LIR:
```
t266 =    CNS_INT   int    -1 REG x2 $4b
        /--*  t266   int
t540 = *  RELOAD    int    REG x1
        /--*  t258   long
        +--*  t259   long
        *  CMP       void   REG NA
        /--*  t540   int
t468 = *  CINCCC    int    cond=SGE REG x1
```

that gets converted to the following sequence of assembly

```
movn    w2, #0
str     w2, [fp, #0x1C]	// [TEMP_01]
ldr     x2, [fp, #0x68]	// [V38 tmp31]
cmp     x2, x1
ldr     w1, [fp, #0x1C]	// [TEMP_01]
cinc    w1, w1, ge
```